From 8a9f64d2aa8c5835d76aaaf5a49a1f51b910d532 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Fri, 30 Aug 2013 19:20:49 +0200
Subject: [PATCH 123/127] wined3d: Fence the destination texture in
 update_texture
Reply-To: wine-devel <wine-devel@winehq.org>

---
 dlls/wined3d/cs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/wined3d/cs.c b/dlls/wined3d/cs.c
index a7b307f..85567b7 100644
--- a/dlls/wined3d/cs.c
+++ b/dlls/wined3d/cs.c
@@ -2211,6 +2211,7 @@ static UINT wined3d_cs_exec_update_texture(struct wined3d_cs *cs, const void *da
     context_release(context);
 
     wined3d_resource_dec_fence(&op->src->resource);
+    wined3d_resource_dec_fence(&op->dst->resource);
 
     return sizeof(*op);
 }
@@ -2226,6 +2227,7 @@ void wined3d_cs_emit_update_texture(struct wined3d_cs *cs, struct wined3d_textur
     op->dst = dst;
 
     wined3d_resource_inc_fence(&op->src->resource);
+    wined3d_resource_inc_fence(&op->dst->resource);
 
     cs->ops->submit(cs, sizeof(*op));
 }
-- 
1.8.1.5

